[Python] -Wstringop-overflow gcc 8 warning

Seems GCC 8 emits a new warning -Wstringop-overflow enabled by -Wall:
```
common_modulePYTHON_wrap.cxx:34313:16: warning: ‘char* strncpy(char*, const char*, size_t)’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
         strncpy(gv->name,name,size);
         ~~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/devel/project/build/python/src/CMakeFiles/_common.dir/common_modulePYTHON_wrap.cxx:34310:27: note: length computed here
       size_t size = strlen(name)+1;
                     ~~~~~~^~~~~~
```

It comes from Lib/python/pyinit.swg:
```
SWIGINTERN void
SWIG_Python_addvarlink(PyObject *p, char *name, PyObject *(*get_attr)(void), int (*set_attr)(PyObject *p)) {
  swig_varlinkobject *v = (swig_varlinkobject *) p;
  swig_globalvar *gv = (swig_globalvar *) malloc(sizeof(swig_globalvar));
  if (gv) {
    size_t size = strlen(name)+1;
    gv->name = (char *)malloc(size);
    if (gv->name) {
      strncpy(gv->name,name,size);
      gv->get_attr = get_attr;
      gv->set_attr = set_attr;
      gv->next = v->vars;
    }
  }
  v->vars = gv;
}
```
Seems it's a false positive, perhaps we could use memcpy ?

--- a/Lib/python/pyinit.swg
+++ b/Lib/python/pyinit.swg
@@ -215,7 +215,7 @@ SWIG_Python_addvarlink(PyObject *p, char *name, PyObject *(*get_attr)(void), int
     size_t size = strlen(name)+1;
     gv->name = (char *)malloc(size);
     if (gv->name) {
-      strncpy(gv->name,name,size);
+      memcpy(gv->name, name, size);
       gv->get_attr = get_attr;
       gv->set_attr = set_attr;
       gv->next = v->vars;

